src/main/java/com/google/firebase/database/core/ThreadPoolEventTarget.java

```diff
@@ -92,8 +96,12 @@ public void uncaughtException(Thread t, Throwable e) {
     synchronized (this) {
       delegate = exceptionHandler;
     }
-    if (delegate != null) {
-      delegate.uncaughtException(t, e);
+    try {
```

Comment:
Nit: I would wrap all of this in the try/finally to reduce the number of braces back to back.